gammapy/time/lomb_scargle.py

```diff
@@ -0,0 +1,237 @@
+    - ``period`` (`float`) -- Location of the highest periodogram peak
+    - ``significance`` (`float`) or (`~numpy.ndarray`) -- Significance of ``period`` under the specified significance criterion. If the significance criterion is not defined, all significance criteria are used and their respective significance for the period is returned
+    - ``swf`` (`~numpy.ndarray`) -- Spectral window function
+
+    Parameters
+    ----------
+    time : `~numpy.ndarray`
+        Time array of the light curve
+    flux : `~numpy.ndarray`
+        Flux array of the light curve
+    flux_err : `~numpy.ndarray`
+        Flux error array of the light curve
+    dt : `float`
+        desired resolution of the periodogram and the window function
+    criterion : `string`
+        significance criterion
+
+        - ``pre`` for pre-defined beta distribution (see Schwarzenberg-Czerny (1998))
+        - ``cvm`` for Cramer-von-Mises distance minimisation (see Thieler et at. (2016))
+        - ``nll`` for negative logarithmic likelihood minimisation
+        - ``boot`` for bootstrap-resampling (see Süveges (2012) and `astroML.time_series.lomb_scargle_bootstrap <http://www.astroml.org/modules/generated/astroML.time_series.lomb_scargle_bootstrap.html>`_)
+    
+    n_bootstraps : `float`
+        Number of bootstraps resampling
+        
+    Returns
+    -------
+    results : `~collections.OrderedDict`
+        Results dictionary (see description above).
+
+    References
+    ----------
+    .. [1] Lomb (1976), "Least-squares frequency analysis of unequally spaced data", 
+       `Link <https://link.springer.com/article/10.1007%2FBF00648343?LI=true>`_
+    .. [2] Scargle (1982), "Studies in astronomical time series analysis. II - Statistical aspects of spectral analysis of unevenly spaced data",
+       `Link <http://articles.adsabs.harvard.edu/full/1982ApJ...263..835S>`_
+    .. [3] Schwarzenberg-Czerny (1998), "The distribution of empirical periodograms: Lomb-Scargle and PDM spectra",
+       `Link <https://academic.oup.com/mnras/article/301/3/831/1038387/The-distribution-of-empirical-periodograms-Lomb>`_
+    .. [4] Thieler et at. (2016), "RobPer: An R Package to Calculate Periodograms for Light Curves Based on Robust Regression",
+       `Link <https://www.jstatsoft.org/article/view/v069i09>`_
+    .. [5] Süveges (2012), "False Alarm Probability based on bootstrap and extreme-value methods for periodogram peaks",
+       `Link <https://www.researchgate.net/profile/Maria_Sueveges/publication/267988824_False_Alarm_Probability_based_on_bootstrap_and_extreme-value_methods_for_periodogram_peaks/links/54e1ba3a0cf2953c22bb222a.pdf>`_
+    """
+    # set up lomb-scargle-algorithm
+    freq = _freq_grid(time, dt)
+    psd_data = LombScargle(time, flux, flux_error).power(freq)
+
+    # find period with highest periodogram peak
+    psd_best_period = np.max(psd_data)
+    best_period = 1. / freq[np.argmax(psd_data)]
+
```

Comment:
My suggestion would be to change the selection of criterion as follows:
```
criteria : list of str
   Select which significance methods you'd like to run (by default all are run)
   Available: {'pre', 'cvm', 'nll', 'boot'}
```
and then to use `criteria=None` as the default and do this in the function:
```
if criteria is None:
    criteria = ['pre', 'cvm', 'nll', 'boot']

significance = OrderedDict()
if 'pre' in criteria:
     significance['pre'] = _significance_pre(time, freq, psd_best_period)
if 'cvm' in criteria:
    significance['pre'] =  _significance_cvm(time, freq, psd_data, psd_best_period)
```
Then the user would have the full power to select exactly the list of criteria they want.
And the results would have names, whereas currently one has to access by index e.g. `significance[2]` to access "cvm"w which is hard to read code.